### ISSUE TITLE
Lots of TypeErrors on startup and operation, window goes blank when given a model to render

### ISSUE THREAD

**Issue by jt6186** [jimy-byerley/uimadcad#24]
I'm getting a bunch of python TypeErrors output to my console when I run uimadcad and when it tries to render bearing.py the whole window goes black.  Both uimadcad and pymadcad were installed using PIP on my Linux Mint system. 

If I just run python3 bearing.py I get a window with the rendering and no errors output.

My Python3 version is 3.10.12

Error messages as follows:

At startup 

 File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 247, in _updateposes
    if obj != 'return' and isinstance(disp, (Solid.display, madcad.rendering.Group)):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1174, in paintGL
    self.render()
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1066, in render
    ViewCommon.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 786, in render
    self.scene.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 576, in render
    func(view)
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 247, in _updateposes
    if obj != 'return' and isinstance(disp, (Solid.display, madcad.rendering.Group)):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1174, in paintGL
    self.render()
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1066, in render
    ViewCommon.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 786, in render
    self.scene.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 576, in render
    func(view)
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 247, in _updateposes
    if obj != 'return' and isinstance(disp, (Solid.display, madcad.rendering.Group)):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1174, in paintGL
    self.render()
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1066, in render
    ViewCommon.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 786, in render
    self.scene.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 576, in render
    func(view)
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 247, in _updateposes
    if obj != 'return' and isinstance(disp, (Solid.display, madcad.rendering.Group)):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union

----------------------------------------------------------------------------------
When I click into the editor window:

Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1174, in paintGL
    self.render()
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1066, in render
    ViewCommon.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 786, in render
    self.scene.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 576, in render
    func(view)
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 247, in _updateposes
    if obj != 'return' and isinstance(disp, (Solid.display, madcad.rendering.Group)):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union

 I also get QTextCursor::setPosition: Position '-1' out of range messages when moving the mouse around.

----------------------------------------------------------------------------------
When bearing.py (minus the show() at the end) is pasted into the editor:

Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/scriptview.py", line 223, in _cursorPositionChanged
    self._updatezone()
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/scriptview.py", line 253, in _updatezone
    main.active_sceneview.scene.sync()
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 108, in sync
    self.update_solidsets()
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 242, in update_solidsets
    recur(self.displays.values())
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 224, in recur
    if isinstance(disp, Solid.display):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1174, in paintGL
    self.render()
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1066, in render
    ViewCommon.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 786, in render
    self.scene.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 576, in render
    func(view)
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 247, in _updateposes
    if obj != 'return' and isinstance(disp, (Solid.display, madcad.rendering.Group)):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/scriptview.py", line 253, in _updatezone
    main.active_sceneview.scene.sync()
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 108, in sync
    self.update_solidsets()
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 242, in update_solidsets
    recur(self.displays.values())
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 224, in recur
    if isinstance(disp, Solid.display):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 76, in _executed
    self.sync()
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 108, in sync
    self.update_solidsets()
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 242, in update_solidsets
    recur(self.displays.values())
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 224, in recur
    if isinstance(disp, Solid.display):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1174, in paintGL
    self.render()
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1066, in render
    ViewCommon.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 786, in render
    self.scene.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 576, in render
    func(view)
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 247, in _updateposes
    if obj != 'return' and isinstance(disp, (Solid.display, madcad.rendering.Group)):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1174, in paintGL
    self.render()
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 1066, in render
    ViewCommon.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 786, in render
    self.scene.render(self)
  File "/home/jt/.local/lib/python3.10/site-packages/madcad/rendering.py", line 576, in render
    func(view)
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/sceneview.py", line 247, in _updateposes
    if obj != 'return' and isinstance(disp, (Solid.display, madcad.rendering.Group)):
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union

And then the whole uimadcad window goes black

**Comment by jimy-byerley** [comment]
Hello @jt6186 
This is because `uimadcad`  doesn't support `pymadcad>=0.17` yet
You can use a development version that supports it in the meantime:

```shell
pip install https://github.com/jimy-byerley/uimadcad/archive/kinematic.zip
```

or use a previous version of pymadcad
```shell
pip install uimadcad pymadcad==0.16 numpy==1.26
```

**Comment by jt6186** [comment]
Switching to the dev version fixed the problems with the black window rendering.  However when I clicked on set active solid button I got a similar type error:
Traceback (most recent call last):
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/gui.py", line 759, in set_active_solid
    found = next((disp	for disp in scene_unroll(self.active_sceneview.scene)
  File "/home/jt/.local/lib/python3.10/site-packages/uimadcad/gui.py", line 760, in <genexpr>
    if isinstance(disp, Solid.display) and disp.selected), 
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union

I had first tried reverting the version of pymadcad but it didn't understand what the fillet command was in bearing.py.  I don't really want to use the old version since fillets are the reason why I'm not going with the popular openSCAD.
